contracts/strategies/aura/base/AuraStrategyUL.sol

```diff
@@ -0,0 +1,613 @@
+  bytes32[] public dexes;
+
+  constructor() public BaseUpgradeableStrategyUL() {
+    assert(_AURA_POOLID_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.auraPoolId")) - 1));
+    assert(_BALANCER_POOLID_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.balancerPoolId")) - 1));
+    assert(_DEPOSIT_TOKEN_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositToken")) - 1));
+    assert(_DEPOSIT_RECEIPT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositReceipt")) - 1));
+    assert(_DEPOSIT_ARRAY_INDEX_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositArrayIndex")) - 1));
+    assert(_BALANCER_DEPOSIT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.balancerDeposit")) - 1));
+    assert(_HODL_RATIO_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlRatio")) - 1));
+    assert(_HODL_VAULT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlVault")) - 1));
+    assert(_NTOKENS_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.nTokens")) - 1)); 
+  }
+
+  function initializeBaseStrategy(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardPool,
+    uint256 _auraPoolID,
+    bytes32 _balancerPoolID,
+    address _depositToken,
+    uint256 _depositArrayPosition,
+    address _balancerDeposit,
+    address[] memory _poolAssets,
+    uint256 _hodlRatio
+  ) public initializer {
+
+    // calculate profit sharing fee depending on hodlRatio
+    uint256 profitSharingNumerator = 300;
+    if (_hodlRatio >= 3000) {
+      profitSharingNumerator = 0;
+    } 
+    else if (_hodlRatio > 0) {
+      // (profitSharingNumerator - hodlRatio/10) * hodlRatioBase / (hodlRatioBase - hodlRatio)
+      // e.g. with default values: (300 - 1000 / 10) * 10000 / (10000 - 1000)
+      // = (300 - 100) * 10000 / 9000 = 222
+      profitSharingNumerator = profitSharingNumerator.sub(_hodlRatio.div(10)) // subtract hodl ratio from profit sharing numerator
+                                    .mul(hodlRatioBase) // multiply with hodlRatioBase
+                                    .div(hodlRatioBase.sub(_hodlRatio)); // divide by hodlRatioBase minus hodlRatio
+    }
+
+    BaseUpgradeableStrategyUL.initialize(
+      _storage,
+      _underlying,
+      _vault,
+      _rewardPool,
+      weth,
+      profitSharingNumerator,  // profit sharing numerator
+      1000, // profit sharing denominator
+      true, // sell
+      0, // sell floor
+      12 hours, // implementation change delay
+      address(0x7882172921E99d590E097cD600554339fBDBc480) //UL Registry
+    );
+
+    rewardTokens = new address[](0);
+    poolAssets = _poolAssets;
+
+    address _lpt;
+    address _depositReceipt;
+    (_lpt,_depositReceipt,,,,) = IAuraBooster(booster).poolInfo(_auraPoolID);
+    require(_lpt == underlying(), "Pool Info does not match underlying");
+    require(1 < _poolAssets.length && _poolAssets.length < 5, "_nTokens should be 2, 3 or 4");
```

Comment:
I don't think there's a need for this check with Balancer. We had it for Curve, as we did not have interface contracts added for other amounts of tokens in the pool. For Balancer I don't think it matters, as we don't need to import different interfaces.